[RDY] Clear prompt_for_number messages

glacambre opened the pull request.

bfredl commented on src/nvim/misc1.c:
```diff
@@ -798,6 +798,9 @@ int prompt_for_number(int *mouse_used)
   }
   State = save_State;
+
+  ui_call_msg_clear();
```
too early, doing something equivalent to `echo inputlist(["aa","bb"]) + inputlist(["cc", "dd"])` should not hide the message after first input. If the issue is specific to spell better fix it in `spell_suggest()`

glacambre replied:
As far as I could tell, `echo inputlist(["aa", "bb"]) + inputlist(["cc", "dd"])` had the same behavior with and without my patch.

I've changed it and forced-pushed your suggestion anyway :).

glacambre committed:
Clear prompt_for_number messages

This fixes issues in GUIs:
https://github.com/akiyosi/goneovim/issues/94
https://github.com/glacambre/firenvim/issues/448

bfredl commented on src/nvim/spell.c:
```diff
@@ -2889,6 +2890,9 @@ void spell_suggest(int count)
     // Ask for choice.
     selected = prompt_for_number(&mouse_used);
+
+    ui_call_msg_clear();
```
one more thing, we want `if ui_has(kUiMessages)` around it :)

glacambre replied:
Should I also fix the linting issues around my patch or leave it alone?

bfredl replied:
yes, if it is not too much :)

bfredl commented on test/functional/ui/messages_spec.lua:
```diff
@@ -795,6 +795,51 @@ describe('ui/ext_messages', function()
+    ]], messages={
+      {content = { { 'Change "helllo" to:\n 1 "Hello"\n 2 "Hallo"\n 3 "Helli"\nType number and <Enter> or click with mouse (empty cancels): ' } }, kind = ""}
+    }, attr_ids = {
+      r = {special = Screen.colors.Red, undercurl = true},
```
add this to `screen:set_default_attr_ids{}` in `before_each` instead.

bfredl commented on test/functional/ui/messages_spec.lua:
```diff
@@ -795,6 +795,51 @@ describe('ui/ext_messages', function()
+      {1:~                        }|
+      {1:~                        }|
+      {1:~                        }|
+    ]], messages={}}
```
empty messages key can be removed (we always assert against unexpected ext_ taste)

glacambre commented:
Thank you for taking the time to review this and all the patient advice! :)
